indy_node/server/request_handlers/domain_req_handlers/context_handler.py

```diff
@@ -0,0 +1,134 @@
+from indy_common.authorize.auth_actions import AuthActionAdd, AuthActionEdit
+from indy_common.authorize.auth_request_validator import WriteRequestValidator
+
+from indy_common.constants import SET_CONTEXT, META
+
+from indy_common.req_utils import get_write_context_name, get_write_context_version, get_txn_context_name, \
+    get_txn_context_version, get_txn_context_data, get_txn_context_meta
+from indy_common.state.state_constants import MARKER_CONTEXT
+from plenum.common.constants import DOMAIN_LEDGER_ID, DATA
+from plenum.common.exceptions import InvalidClientRequest
+
+from plenum.common.request import Request
+from plenum.common.txn_util import get_request_data, get_from, get_seq_no, get_txn_time
+from plenum.server.database_manager import DatabaseManager
+from plenum.server.request_handlers.handler_interfaces.write_request_handler import WriteRequestHandler
+from plenum.server.request_handlers.utils import encode_state_value
+
+from re import findall
+
+URI_REGEX = r'^(?P<scheme>\w+):(?:(?:(?P<url>//[.\w]+)(?:(/(?P<path>[/\w]+)?)?))|(?:(?P<method>\w+):(?P<id>\w+)))'
+
+
+class ContextHandler(WriteRequestHandler):
+
+    def __init__(self, database_manager: DatabaseManager,
+                 write_req_validator: WriteRequestValidator):
+        super().__init__(database_manager, SET_CONTEXT, DOMAIN_LEDGER_ID)
+        self.write_req_validator = write_req_validator
+
+    def static_validation(self, request: Request):
+        self._validate_request_type(request)
+        meta = request.operation['meta']
+        validate_meta(meta)
+        data = request.operation['data']
+        validate_data(data)
+
+    def dynamic_validation(self, request: Request):
+        # we can not add a Context with already existent NAME and VERSION
+        # since a Context needs to be identified by seqNo
```

Comment:
It would be great to create a Design doc / HIPE describing how reach schema objects are related to each other. 
Does it use `seqNo` or other IDs for referencing? 

Currently Schema is referenced by ClaimDef via `seqNo` on Indy-Node level, but ClaimDef is referenced by RevocRegDef via generic ID mechanism which is currently equal to ClaimDef's key in the state trie, that is unique ClaimDef identifier allowing to potentially update referenced ClaimDef (state can be considered as a dict accumulates all the changes for an object).
On libindy level everything is supposed to be referenced via IDs (including referencing a Schema from a ClaimDef).